File: algorithms/javascript/selection_sort.js
function indexOfMin(list) {
    var minIndex = 0;
    for(var i = 0; i < list.length; i++) {
        if (list[i] < list[minIndex]) {
            minIndex = i;
        }
    }
    return minIndex;
}
function selectionSort(list) {
    var sortedList = [];
    while(list.length > 0) {
        var indexToMove = indexOfMin(list);
        sortedList.push(list.splice(indexToMove, 1)[0]);
    }
    return sortedList;
}

const testValues = [29, 100, 1, 2, 57, 28, 88, 3, 50, 67, 37, 1, 32, 20];
var sorted = selectionSort(testValues);
console.log(sorted);

File: algorithms/python/selection_sort.py
def selection_sort(values):
    sorted_list = []
    for _idx in range(0, len(values)):
        index_to_move = index_of_min(values)
        sorted_list.append(values.pop(index_to_move))

    return sorted_list


def index_of_min(values):
    min_index = 0
    for i in range(1, len(values)):
        if values[i] < values[min_index]:
            min_index = i

    return min_index


numbers = [14, 3, 1, 9, 10, 4, 6, 2, 21, 13]
print(selection_sort(numbers))
